### PR TITLE
verify external page

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
add a page to verify email from blog.kyberswap.com
add count down 5s after verify
test link: https://blog.kyberswap.com/?test_email=true

### CODE REVIEW

File: src/pages/Verify/index.tsx [hunk 1]
```diff
@@ -40,41 +41,68 @@ enum STATUS {
...
+  refTime.current = time
+  const handleCountDown = useCallback(() => {
+    const time = refTime.current
+    if (time > 0) {
+      setTime(time => time - 1)
+      return
+    }
+    refTimeoutCountDown.current && clearInterval(refTimeoutCountDown.current)
+    if (isVerifyExternal) {
+      if (redirectUrl) window.location.href = redirectUrl
```
Comment (nguyenhoaidanh) [inline src/pages/Verify/index.tsx:76]: validate at validateUrl()

File: src/pages/Verify/index.tsx [hunk 2]
```diff
@@ -40,41 +41,68 @@
...
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/pages/Verify/index.tsx:76]: validate at validateUrl()

[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/pages/Verify/index.tsx:76]: validate at validateUrl()

[the remark this answers is not in the document]
Comment (hungdoansy) [reply src/pages/Verify/index.tsx:76]: Why not use react-router here?

**Review by hungdoansy — CHANGES_REQUESTED** [review]

File: src/pages/Verify/index.tsx [hunk 3]
```diff
@@ -40,41 +41,68 @@ enum STATUS {
   ERROR = 'Error occur, please try again.',
 }
 
+function validateUrl(url: string) {
+  if (!url) return ''
+  const whitelistDomain = ['https://blog.kyberswap.com']
+  try {
+    if (whitelistDomain.some(e => new URL(e).host === new URL(url).host)) return url
```
Comment (hungdoansy) [inline src/pages/Verify/index.tsx:48]: This is unsafe, `new URL('asdasd')` will throw an error

Comment (nguyenhoaidanh) [reply src/pages/Verify/index.tsx:48]: added try catch

Comment (hungdoansy) [reply src/pages/Verify/index.tsx:48]: Ah ok, I see the try catch. But why not just comparing with `startsWith`?

Comment (nguyenhoaidanh) [reply src/pages/Verify/index.tsx:48]: for sure , it will work for both
http://blog.kyberswap.com
https://blog.kyberswap.com

**Review by hungdoansy — DISMISSED** [review]

**Review by hungdoansy — APPROVED** [review]

**Pull request merged**